sycl/test-e2e/Matrix/get_coord_int8_matB_impl.hpp

```diff
@@ -93,69 +92,79 @@ wi [1,0] -->    i=0, [8, 0]
-           const auto sg_starty = global_idy - spmd_item.get_local_id(1);
-
-           sycl::sub_group sg = spmd_item.get_sub_group();
-
-           joint_matrix<sub_group, int8_t, use::b, TK, TN,
-                        ext::intel::experimental::matrix::layout::packed>
-               sub_b;
-
-           joint_matrix_load(
-               sg, sub_b,
-               accB.template get_multi_ptr<access::decorated::no>() +
-                   (sg_startx * (TK / VF) * N) + sg_starty / SG_SZ * TN * VF,
-               N);
-
-           int32_t sum_local_cols[N] = {0};
-           auto wiData =
-               sycl::ext::intel::experimental::matrix::get_wi_data(sg, sub_b);
-
-           // each WI calculates local sum of cols
-           for (int i = 0; i < wiData.length(); ++i) {
-             // get the index of the element in the submatrix
-             auto dataItem = wiData[i];
-             auto [row, col] = dataItem.get_coord();
-             size_t global_index = col + global_idy / SG_SZ * TN * VF;
-             sum_local_cols[global_index] += dataItem;
-           }
-
-           for (int i = 0; i < N; i++) {
-             sum_local_cols[i] =
-                 reduce_over_group(sg, sum_local_cols[i], sycl::plus<>());
-             if (global_idy % SG_SZ == 0)
-               atomic_fetch_add(v[i], sum_local_cols[i]);
-           }
-         }); // parallel for
+     cgh.parallel_for(r, [=](nd_item<2> spmd_item) [[intel::reqd_sub_group_size(
+                             SG_SZ)]] {
+       const auto global_idx = spmd_item.get_global_id(0);
+       const auto global_idy = spmd_item.get_global_id(1);
+       const auto sg_startx = global_idx - spmd_item.get_local_id(0);
+       const auto sg_starty = global_idy - spmd_item.get_local_id(1);
+
+       sycl::sub_group sg = spmd_item.get_sub_group();
+
+       joint_matrix<sub_group, int8_t, use::b, TK, TN,
+                    ext::intel::experimental::matrix::layout::packed>
+           sub_b;
+
+       joint_matrix_load(sg, sub_b,
+                         accB.template get_multi_ptr<access::decorated::no>() +
+                             (sg_startx * (TK / VF) * N * VF) +
+                             sg_starty / SG_SZ * TN * VF,
+                         N * VF);
+
+       int32_t sum_local_cols[N] = {0};
+       auto wiData =
+           sycl::ext::intel::experimental::matrix::get_wi_data(sg, sub_b);
+
+       // each WI calculates local sum of cols
+       for (int i = 0; i < wiData.length(); ++i) {
+         // get the index of the element in the submatrix
+         auto dataItem = wiData[i];
+         // the coordinates returned are in the logical range [K,N]
+         // If users want to retrieve the VNNIed coordinates, they can be
+         // obtained using
+         // colVNNI = col/VF
+         // rowVNNI = row*VF
+         auto [row, col] = dataItem.get_coord();
+         size_t global_index = col + global_idy / SG_SZ * TN; // * VF;
```

Comment:
seems this comment is unnecessary
```suggestion
         size_t global_index = col + global_idy / SG_SZ * TN;
```